source/extensions/filters/network/redis_proxy/command_splitter_impl.h

```diff
@@ -90,6 +90,48 @@ class SplitRequestBase : public SplitRequest, public Logger::Loggable<Logger::Id
   Stats::TimespanPtr command_latency_;
 };
 
+/**
+ * NoKeyAllhostRequest is a base class for commands which has no keys but has to be applied on all primaries.
+ */
+class NoKeyAllPrimaryRequest : public SplitRequestBase {
```

Comment:
let's just call this `NoKeyRequest` or perhaps `MultipleNodeRequest`.  the decision of whom to send it to (all primaries in this case), can be an implementation detail.

Reply:
ok sure vijay

Reply:
this has ben renamed to AdministrationRequest as they are used for handling all admin related requests